Lib/test/test_xml_etree.py

```diff
@@ -2693,6 +2694,41 @@ def __eq__(self, o):
         e.extend([ET.Element('bar')])
         self.assertRaises(ValueError, e.remove, X('baz'))
 
+    def test_remove_with_clear_children(self):
+        # See: https://github.com/python/cpython/issues/126033
+
+        class X(ET.Element):
+            def __eq__(self, o):
+                root.clear()
+                return False
+
+        for foo_type, rem_type in [(X, ET.Element), (ET.Element, X)]:
```

Comment:
What is the difference between these cases? Why testing only one case is not enough?

Reply:
I wanted to test when removal happens at the beginning or at the end of the loop.